kedro/framework/cli/catalog.py

```diff
@@ -207,3 +207,51 @@ def rank_catalog_factories(metadata: ProjectMetadata, env):
         click.echo(yaml.dump(list(catalog_factories.keys())))
     else:
         click.echo("There are no dataset factories in the catalog.")
+
+
+@catalog.command("resolve")
+@env_option
+@click.pass_obj
+def resolve_patterns(metadata: ProjectMetadata, env):
+    """Resolve catalog factories against pipeline datasets"""
+
+    session = _create_session(metadata.package_name, env=env)
+    context = session.load_context()
+
+    data_catalog = context.catalog
+    catalog_config = context.config_loader["catalog"]
+
+    catalog_config = {
```

Comment:
It's a bit confusing that these variables have the exact name. In general it is best practice not to use the same variable as input and reassign the value in the same step. Maybe the second `catalog_config` can be re-named to `explicit_dataset`. That's what's stored there right?

Reply:
Yep! Changed in [779a286](https://github.com/kedro-org/kedro/pull/2891/commits/779a286f31c317ceb07591ff3c85d70ffcf40dd6)